airflow_config/scripts/init_airflow_service.py

```diff
@@ -0,0 +1,262 @@
+# Airflow init script for Dockerfile ENTRYPOINT
+from logging import error
+import os
+import sys
+from cryptography.fernet import Fernet
+import time
+
+###################################
+# Global defaults and back-compat #
+###################################
+# Airflow global default
+# Get env config
+AIRFLOW_HOME = os.getenv("AIRFLOW_HOME", "/usr/local/airflow")
+if os.getenv("EXECUTOR") is None:
+    AIRFLOW__CORE__EXECUTOR = "SecuentialExecutor"
+else:
+    AIRFLOW__CORE__EXECUTOR = os.getenv("EXECUTOR") + "Executor"
+AIRFLOW__CORE__LOAD_EXAMPLES = os.getenv("AIRFLOW__CORE__LOAD_EXAMPLES", "0")
+AIRFLOW__CORE__DAGS_ARE_PAUSED_AT_CREATION = os.getenv(
+    "AIRFLOW__CORE__DAGS_ARE_PAUSED_AT_CREATION", "0"
+)
+AIRFLOW__API__AUTH_BACKEND = os.getenv(
+    "AIRFLOW__API__AUTH_BACKEND", "airflow.api.auth.backend.basic_auth"
+)
+AIRFLOW__CORE__FERNET_KEY = os.getenv("FERNET_KEY", Fernet.generate_key().decode())
+AIRFLOW_USER = os.getenv("AIRFLOW_USER", "admin")
+AIRFLOW_FIRSTNAME = os.getenv("AIRFLOW_FIRSTNAME", "admin")
+AIRFLOW_LASTNAME = os.getenv("AIRFLOW_LASTNAME", "admin")
+AIRFLOW_ROLE = os.getenv("AIRFLOW_ROLE", "Admin")
+AIRFLOW_PWD = os.getenv("AIRFLOW_PWD", "admin")
+AIRFLOW_USER_EMAIL = os.getenv("AIRFLOW_USER_EMAIL", "admin@example.com")
+CORNFLOW_SERVICE_USER = os.getenv("CORNFLOW_SERVICE_USER", "serviceuser@cornflow.com")
+CORNFLOW_SERVICE_PWD = os.getenv("CORNFLOW_SERVICE_PWD", "servicecornflow1234")
+AIRFLOW_LDAP_ENABLE = os.getenv("AIRFLOW_LDAP_ENABLE", "False")
+CUSTOM_SSH_HOST = os.getenv("CUSTOM_SSH_HOST")
+
+# update os environ
+os.environ["AIRFLOW_HOME"] = AIRFLOW_HOME
+os.environ["AIRFLOW__CORE__EXECUTOR"] = AIRFLOW__CORE__EXECUTOR
+os.environ["AIRFLOW__CORE__LOAD_EXAMPLES"] = AIRFLOW__CORE__LOAD_EXAMPLES
+os.environ[
+    "AIRFLOW__CORE__DAGS_ARE_PAUSED_AT_CREATION"
+] = AIRFLOW__CORE__DAGS_ARE_PAUSED_AT_CREATION
+os.environ["AIRFLOW__API__AUTH_BACKEND"] = AIRFLOW__API__AUTH_BACKEND
+os.environ["AIRFLOW_CORE__FERNET_KEY"] = AIRFLOW__CORE__FERNET_KEY
+os.environ["AIRFLOW_USER"] = AIRFLOW_USER
+os.environ["AIRFLOW_FIRSTNAME"] = AIRFLOW_FIRSTNAME
+os.environ["AIRFLOW_LASTNAME"] = AIRFLOW_LASTNAME
+os.environ["AIRFLOW_ROLE"] = AIRFLOW_ROLE
+os.environ["AIRFLOW_PWD"] = AIRFLOW_PWD
+os.environ["AIRFLOW_USER_EMAIL"] = AIRFLOW_USER_EMAIL
+os.environ["CORNFLOW_SERVICE_USER"] = CORNFLOW_SERVICE_USER
+os.environ["CORNFLOW_SERVICE_PWD"] = CORNFLOW_SERVICE_PWD
+os.environ["AIRFLOW_LDAP_ENABLE"] = AIRFLOW_LDAP_ENABLE
```

Comment:
moreover, could we have something like:

```python

env_vars = [
("AIRFLOW_HOME", "/usr/local/airflow"), 
("AIRFLOW__CORE__EXECUTOR", "SecuentialExecutor"), 
("AIRFLOW__CORE__LOAD_EXAMPLES", "0"),
 ...
 ]

for name, default in env_vars:
    os.environ[name] = os.getenv(name, default)
```

What to they do in airflow? I know they have some python code to parse env vars.

Reply:
It certainly seems even cleaner to me. These environment variables must be exported precisely so that Airflow takes them into account in its initial configuration. Some I make local variables because we use them later in this script. Although I suppose there is no problem in reading the environment variable at all times.